consensus/state_sync/module.go

```diff
@@ -24,41 +43,140 @@ var (
 type stateSync struct {
 	bus    modules.Bus
 	logger *modules.Logger
+
+	// metadata responses received from peers are collected in this channel
+	metadataReceived chan *typesCons.StateSyncMetadataResponse
+
+	committedBlocksChannel chan uint64
 }
 
-func CreateStateSync(bus modules.Bus, options ...modules.ModuleOption) (modules.Module, error) {
+func Create(bus modules.Bus, options ...modules.ModuleOption) (modules.Module, error) {
 	return new(stateSync).Create(bus, options...)
 }
 
 func (*stateSync) Create(bus modules.Bus, options ...modules.ModuleOption) (modules.Module, error) {
-	m := &stateSync{}
+	m := &stateSync{
+		metadataReceived:       make(chan *typesCons.StateSyncMetadataResponse, metadataChannelSize),
+		committedBlocksChannel: make(chan uint64, committedBlocsChannelSize),
+	}
+	m.logger = logger.Global.CreateLoggerForModule(m.GetModuleName())
```

Comment:
Is it important to create the logger before applying options and registering module?

Reply:
We do not have strict guidelines for this right now and it doesn't really matter at the moment.

[### Add a logger to the module](https://github.com/pokt-network/pocket/blob/77dc729951c6cae4999297d97dc219b106ff0cb7/shared/modules/doc/README.md?plain=1#L494)

<img width="1322" alt="Screenshot 2023-07-25 at 2 25 54 PM" src="https://github.com/pokt-network/pocket/assets/1892194/67b77a52-6049-4d61-808a-d726e4d14656">

If I had to decide on guidelines right now, I would say:
1. Apply options
2. Create a logger.